### PR TITLE
fix(clock): fix pauseAt to arrive at wall time

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/injected/clock.ts [hunk 1]
```diff
@@ -99,40 +99,41 @@ export class ClockController {
...
-    this._innerSetTime(time);
+    this._innerSetTime(asWallTime(time));
   }
 
-  setSystemTime(time: number) {
+  setSystemTime(time: number): number {
     this._replayLogOnce();
-    this._innerSetTime(time);
+    this._innerSetTime(asWallTime(time));
+    return time - this._now.time;
```
Comment (dgozman) [inline packages/playwright-core/src/server/injected/clock.ts:108]: It looks like this value if always zero?

File: packages/playwright-core/src/server/injected/clock.ts [hunk 2]
```diff
@@ -282,10 +285,12 @@ export class ClockController {
 
     this._advanceNow(timer.callAt);
 
-    if (timer.type === TimerType.Interval)
-      this._timers.get(timer.id)!.callAt += timer.delay;
-    else
+    if (timer.type === TimerType.Interval) {
+      const callAt = this._timers.get(timer.id)!.callAt;
```
Comment (dgozman) [inline packages/playwright-core/src/server/injected/clock.ts:289]: I wonder whether `this._timers.get(timer.id) === timer`?

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/server/injected/clock.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/injected/clock.ts:108]: Yeah, I don't need this return value, it is leftovers, removed.

File: packages/playwright-core/src/server/injected/clock.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/injected/clock.ts:289]: I'm being stupid - doing a local refactoring w/o looking at context.

**Pull request merged**